sdb/commands/linux/internal/slub_helpers.py

```diff
@@ -0,0 +1,108 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+from typing import Iterable
+
+import drgn
+from drgn.helpers.linux.list import list_for_each_entry
+
+
+def is_root_cache(obj: drgn.Object) -> bool:
+    return obj.memcg_params.root_cache.value_() == 0x0
```

Comment:
is there a canonical way to check or assert that the drgn.Object is of the expected C type?  Or do we just rely on "duck typing"?  If nothing else, we could declare the expected C type via a comment.

Reply:
Added asserts to check the name of the C type for those functions is what we expect:
```
+    assert obj.type_.type_name() == 'struct kmem_cache *'
```